Sming/SmingCore/Network/HttpClient.cpp

```diff
@@ -13,25 +13,21 @@
 #include "HttpClient.h"
 #include "Data/Stream/FileStream.h"
 
-HashMap<String, HttpClientConnection*> HttpClient::httpConnectionPool;
+ObjectMap<String, HttpClientConnection> HttpClient::httpConnectionPool;
 
-/* Low Level Methods */
 bool HttpClient::send(HttpRequest* request)
 {
 	String cacheKey = getCacheKey(request->uri);
 
-	HttpClientConnection* connection = nullptr;
+	auto& connection = httpConnectionPool[cacheKey];
 
-	int i = httpConnectionPool.indexOf(cacheKey);
-	if(i >= 0) {
+	if(connection != nullptr) {
 		// Check existing connection
-		connection = httpConnectionPool.valueAt(i);
 		if(connection->getConnectionState() > eTCS_Connecting && !connection->isActive()) {
 			debug_d("Removing stale connection: State: %d, Active: %d", connection->getConnectionState(),
 					connection->isActive());
 			delete connection;
```

Comment:
I wonder, is it necessary to delete the connection? Could we just reset it, perhaps?